FormidableLabs/victory
# Modularize calculated props helpers

This adds custom hooks that can be used to calculate Victory props inside a context provider component.

`useDomain` - Accepts domain props and returns a domain for the data formatted as `[min, max]` (currently excludes domain padding logic)
`useRange` - Returns a range using height, width, and padding
`useScale` - Returns the base scale (no domain or range) from a string or scale prop
`useData` - Returns normalized data 

This also includes a basic `VictoryProvider` component that stores these calculated values, and a demo component that uses this provider. 

The Victory bar demo component demonstrates how we can use these values for rendering data, but it does not include events, styling, or animations. 

To test:
- Change the export in `victory-bar/index.js` from `victory-bar` to `victory-bar-beta`. 
- Skip the `describe` block in `victory-bar.test.js` that contains events and run `yarn nps "jest victory-bar". This component should pass all other tests for victory bar.
